### PR TITLE
[HttpKernel] don't call getTrustedHeaderName() if possible

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://travis-ci.org/symfony/symfony/jobs/235008102 (failing tests of #22863)
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Fragment/InlineFragmentRenderer.php [hunk 1]
```diff
@@ -119,10 +119,14 @@ protected function createSubRequest($uri, Request $request)
         // Sub-request object will point to localhost as client ip and real client ip
         // will be included into trusted header for client ip
         try {
-            if ($trustedHeaderName = Request::getTrustedHeaderName(Request::HEADER_CLIENT_IP, false)) {
+            if (method_exists(Request::class, 'getTrustedHeaderName') && $trustedHeaderName = Request::getTrustedHeaderName(Request::HEADER_CLIENT_IP, false)) {
                 $currentXForwardedFor = $request->headers->get($trustedHeaderName, '');
 
                 $server['HTTP_'.$trustedHeaderName] = ($currentXForwardedFor ? $currentXForwardedFor.', ' : '').$request->getClientIp();
+            } elseif (Request::HEADER_X_FORWARDED_FOR & Request::getTrustedHeaderSet()) {
```
Comment (stof) [inline src/Symfony/Component/HttpKernel/Fragment/InlineFragmentRenderer.php]: Actually, the non-deprecated API should be tried first, to use it when it is available IMO

Comment (xabbuh) [inline src/Symfony/Component/HttpKernel/Fragment/InlineFragmentRenderer.php]: In fact, I think that we can just fix this in 3.3.

File: src/Symfony/Component/HttpKernel/Fragment/InlineFragmentRenderer.php [hunk 2]
```diff
@@ -119,7 +119,13 @@ protected function createSubRequest($uri, Request $request)
         // Sub-request object will point to localhost as client ip and real client ip
         // will be included into trusted header for client ip
         try {
-            if ($trustedHeaderName = Request::getTrustedHeaderName(Request::HEADER_CLIENT_IP, false)) {
+            $hasTrustedHeaderSet = method_exists(Request::class, 'getTrustedHeaderSet');
```
Comment (stof) [inline src/Symfony/Component/HttpKernel/Fragment/InlineFragmentRenderer.php]: I would avoid this check by bumping the min version of HttpFoundation in HttpKernel

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**